tests/unit/core/components/TestAMInstallButton.js

```diff
@@ -393,19 +393,48 @@ describe(__filename, () => {
     const enable = sinon.spy();
     const install = sinon.spy();
 
-    const root = render({ addon, enable, install });
+    const root = render({
+      addon,
+      enable,
+      install,
+      getAddonEnabledStatus: sinon.stub().returns(false),
+    });
 
     const event = createFakeEvent();
 
     const installButton = root.find('.AMInstallButton-button');
     await installButton.simulate('click', event);
 
-    sinon.assert.calledOnce(install);
+    await sinon.assert.calledOnce(install);
```

Comment:
> @kumar303 would you have an idea to make it better?

This is the first time I've seen `sinon.assert.calledOnce()` used as a promise and I can't find any documentation on this behavior so I don't think we should rely on it. My guess is that it only works by chance in that it waits for one more tick on the event loop. Otherwise, what would cause the `calledOnce()` promise to resolve? It is not clear.

I would like to see these tests rewritten with proper async handling.

The problem is that the click handlers rely on promises. Here is a suggestion that doesn't fully cover hooking up the click handlers but I'd be fine with it since it's a simple change from what you have already and it's probably good enough.

```diff
diff --git a/tests/unit/core/components/TestAMInstallButton.js b/tests/unit/core/components/TestAMInstallButton.js
index 3d1b60693..b8bbf2233 100644
--- a/tests/unit/core/components/TestAMInstallButton.js
+++ b/tests/unit/core/components/TestAMInstallButton.js
@@ -403,9 +403,10 @@ describe(__filename, () => {
     const event = createFakeEvent();
 
     const installButton = root.find('.AMInstallButton-button');
-    await installButton.simulate('click', event);
+    const onClick = installButton.prop('onClick');
+    await onClick(event);
 
-    await sinon.assert.calledOnce(install);
+    sinon.assert.calledOnce(install);
     sinon.assert.calledOnce(enable);
     sinon.assert.calledOnce(event.preventDefault);
     sinon.assert.calledOnce(event.stopPropagation);
```